lib/facter/custom_facts/core/aggregate.rb

```diff
@@ -101,7 +108,22 @@ def options(options)
       #
       # @api private
       def evaluate(&block)
+        if @last_evaluated
+          msg = "Already evaluated #{@name}"
+          msg << " at #{@last_evaluated}" if msg.is_a? String
+          msg << ', reevaluating anyways'
+          log.warn msg
+        end
         instance_eval(&block)
+
+        # Ruby 1.9+ provides the source location of procs which can provide useful
+        # debugging information if a resolution is being evaluated twice. Since 1.8
+        # doesn't support this we opportunistically provide this information.
+        @last_evaluated = if block.respond_to? :source_location
+                            block.source_location.join(':')
+                          else
+                            true
+                          end
```

Comment:
I don't think we need to account for versions of Ruby older than 1.9 since we only claim to support >= 2.5 in the Facter gemspec

Reply:
good point!